### PR TITLE
Cal/collision layers

### PR DESCRIPTION
**Pull request by McCallisterRomer**
-Added first allocator and pool memory resource
-Made changes to ComponentSystem to use new allocator
-Unit tests for new classes
-Added a few entries to config for ComponentSystem

### CODE REVIEW

File: nc/source/alloc/Allocator.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     *  @todo Two allocators templated on the same resource will share that
+     *  resource. Not sure if this is what we want. This could create its own
+     *  resource, or it could take a ptr in c'tor? */
+    template<class Resource>
+        requires MemoryResource<Resource>
+    class FixedTypeAllocator
+    {
+        public:
+            using value_type = Resource::value_type;
+            using propogate_on_container_copy_assignment = std::false_type;
```
Comment (jaremieromer) [inline nc/source/alloc/Allocator.h]: propagate

File: nc/source/alloc/Allocator.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    template<class Resource>
+    void FixedTypeAllocator<Resource>::clear_memory_resource()
+    {
+        m_memoryResource->free_all();
+    }
+
+    template<class Resource>
+    void FixedTypeAllocator<Resource>::release_memory_resource()
+    {
+        if(!m_memoryResource)
```
Comment (jaremieromer) [inline nc/source/alloc/Allocator.h:76]: Don't we need these checks on all functions here?

File: nc/source/alloc/MemoryResource.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        public:
+            using value_type = T;
+            using storage_type = std::aligned_storage_t<sizeof(T), alignof(T)>;
+
+            StaticPool(size_t maxCount);
+            ~StaticPool() = default;
+            StaticPool(const StaticPool&) = delete;
+            StaticPool& operator=(const StaticPool&) = delete;
+
+            T* allocate(std::size_t n);
```
Comment (jaremieromer) [inline nc/source/alloc/MemoryResource.h]: why just std:: for this size_t? remove or add elsewhere for consistency

File: nc/source/config/config.ini [hunk 4]
```diff
@@ -1,6 +1,11 @@
 [project]
 project_name=NcEngine Samples
 log_file_path=Diagnostics.log
+[memory]
+max_colliders=50000
```
Comment (jaremieromer) [inline nc/source/config/config.ini:5]: How do we fail if a user were to set this to a number such that their system would run out of memory? ie (500,000,000,000)

File: nc/source/ecs/ComponentSystem.h [hunk 5]
```diff
@@ -1,169 +1,109 @@
 #pragma once
```
Comment (jaremieromer) [inline nc/source/ecs/ComponentSystem.h:1]: So much less code in this file now!

File: nc/source/physics/ColliderSystem.h [hunk 6]
```diff
@@ -12,7 +12,7 @@ namespace nc::physics
     {
         public:
             /** @todo either make this dynamically sized or set this from config */
```
Comment (jaremieromer) [inline nc/source/physics/ColliderSystem.h]: You can remove this @todo now. Nice!

Comment (McCallisterRomer) [inline nc/source/physics/ColliderSystem.h]: delete comment

**Review by jaremieromer — APPROVED** [review]
!!!

**Pull request merged**